src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -140,13 +143,25 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     var platform =
         CloudPlatformWrapper.of(sourceDataset.getDatasetSummary().getStorageCloudPlatform());
 
+    boolean isTdrBillingProfile =
+        inputParameters.get(JobMapKeys.TDR_BILLING_PROFILE_FALLBACK.getKeyName(), Boolean.class);
+
     // Take out a shared lock on the source dataset, to guard against it being deleted out from
     // under us (for example)
     addStep(new LockDatasetStep(datasetService, datasetId, true), randomBackoffRetry);
 
-    // Make sure this user is authorized to use the billing profile in SAM
-    addStep(
-        new AuthorizeBillingProfileUseStep(profileService, snapshotReq.getProfileId(), userReq));
+    if (isTdrBillingProfile) {
```

Comment:
This now duplicates the code in `DatasetCreateFlight`. If we wanted to avoid this duplication, one approach would be to put both cases inside `AuthorizeBillingProfileUseStep` and have it also accept `rawlsService` and `isTdrBillingProfile`.

Another benefit (or shortcoming, depending on how you look at it) of this refactoring is that changing `AuthorizeBillingProfileUseStep` means that we'd have to look at all cases where this step is used and fix them as part of this work.